runtime/src/iree/hal/command_buffer.h

```diff
@@ -725,8 +685,7 @@ IREE_API_EXPORT iree_status_t iree_hal_command_buffer_push_constants(
 IREE_API_EXPORT iree_status_t iree_hal_command_buffer_push_descriptor_set(
     iree_hal_command_buffer_t* command_buffer,
     iree_hal_pipeline_layout_t* pipeline_layout, uint32_t set,
-    iree_host_size_t binding_count,
-    const iree_hal_descriptor_set_binding_t* bindings);
+    iree_host_size_t binding_count, const iree_hal_buffer_ref_t* bindings);
```

Comment:
I found these diffs in the CTS test usage confusing, since `iree_hal_command_buffer_push_descriptor_set()` no longer takes an argument with `descriptor_set` in the name:
```diff
-  iree_hal_descriptor_set_binding_t descriptor_set_bindings[] = {
+  iree_hal_buffer_ref_t descriptor_set_bindings[] = {
```

Looks like the `iree_hal_descriptor_set_binding_t` struct got merged into `iree_hal_buffer_ref_t`.

How about updating the docstring comment for this function to more directly associate "buffer bindings" with "descriptor sets"? Could also rename the parameter and/or struct.

Reply:
did I just get permission to increase scope? I think I did! ;P
I may do a follow-up that renames `push_descriptor_set` to `push_bindings`. Descriptor set layouts are still used by pipeline layouts but we don't have a HAL concept of a descriptor set resource so it doesn't make much sense beyond matching Vulkan naming.

For now I'll keep it the same but added a description to the comment.

Reply:
Scope creep yay!

SGTM